pkg/asset/installconfig/powervs/metadata.go

```diff
@@ -491,3 +493,105 @@ func leftInContext(ctx context.Context) time.Duration {
+	found := false
+	for _, service := range overrides {
+		if service.Name == name {
+			found = true
+			break
+		}
+	}
+	if !found {
+		return append(overrides, configv1.PowerVSServiceEndpoint{Name: name, URL: url})
+	}
+	return overrides
+}
+
+// CreateVirtualPrivateEndpointGateways checks and creates necessary VPEs in given target VPC for Disconnected install.
+func (m *Metadata) CreateVirtualPrivateEndpointGateways(ctx context.Context, infraID string, region string, vpcID string, subnetID string, groupID string, endpointOverrides []configv1.PowerVSServiceEndpoint) error {
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+
+	client, err := m.client()
+	if err != nil {
+		return err
+	}
+
+	endpoint, err := m.fetchEndpointOverride(endpointOverrides, "IAM")
+	if err != nil {
+		return fmt.Errorf("failed to fetch endpoint override found for IAM: %w", err)
+	}
+	if endpoint == "" {
+		name := fmt.Sprintf("%s-vpe-iam", infraID)
+		iamCrn := "crn:v1:bluemix:public:iam-svcs:global:::endpoint:private.iam.cloud.ibm.com"
+		logrus.Debugf("InfraReady: Ensuring VPE gateway for IAM %v", iamCrn)
+		_, err = client.CreateVirtualPrivateEndpointGateway(ctx, name, vpcID, subnetID, groupID, iamCrn)
+		if err != nil {
+			return fmt.Errorf("failed to create VPE: %w", err)
+		}
+	}
+
+	endpoint, err = m.fetchEndpointOverride(endpointOverrides, "COS")
+	if err != nil {
+		return fmt.Errorf("failed to fetch endpoint override found for COS: %w", err)
+	}
+	if endpoint == "" {
+		name := fmt.Sprintf("%s-vpe-cos", infraID)
+		cosCrn := fmt.Sprintf("crn:v1:bluemix:public:cloud-object-storage:global:::endpoint:s3.direct.%s.cloud-object-storage.appdomain.cloud", region)
+		logrus.Debugf("InfraReady: Ensuring VPE gateway for COS %v", cosCrn)
+		_, err = client.CreateVirtualPrivateEndpointGateway(ctx, name, vpcID, subnetID, groupID, cosCrn)
+		if err != nil {
+			return fmt.Errorf("failed to create VPE: %w", err)
+		}
+	}
+
+	endpoint, err = m.fetchEndpointOverride(endpointOverrides, "DNSServices")
+	if err != nil {
+		return fmt.Errorf("failed to fetch endpoint override found for DNS: %w", err)
+	}
+	if endpoint == "" {
+		name := fmt.Sprintf("%s-vpe-dns", infraID)
+		dnsCrn := "crn:v1:bluemix:public:dns-svcs:global::::"
+		logrus.Debugf("InfraReady: Ensuring VPE gateway for DNS services %v", dnsCrn)
+		_, err = client.CreateVirtualPrivateEndpointGateway(ctx, name, vpcID, subnetID, groupID, dnsCrn)
+		if err != nil {
+			return fmt.Errorf("failed to create VPE: %w", err)
+		}
+	}
+
+	endpoint, err = m.fetchEndpointOverride(endpointOverrides, string(configv1.IBMCloudServiceResourceController))
+	if err != nil {
+		return fmt.Errorf("failed to fetch endpoint override found for RC: %w", err)
+	}
+	if endpoint == "" {
+		name := fmt.Sprintf("%s-vpe-rc", infraID)
+		rcCrn := "crn:v1:bluemix:public:resource-controller:global:::endpoint:private.resource-controller.cloud.ibm.com"
```

Comment:
Nit: Please add this as a constant.

Reply:
Addressed in the latest force-push. PTAL.